### PR TITLE
feat(artifacts): update gitrepo configuration

### PR DESCRIPTION
**Pull request by KathrynLewis**
Git Repo artifacts are now configurable through Halyard, so this removes the directions for configuring through a `clouddriver-local.yml` and adds directions for using Halyard.

### CODE REVIEW

**Comment by 317brian** [comment]
Thanks for putting this together @KathrynLewis!

File: setup/artifacts/gitrepo.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+layout: single
+title:  "Configure a Git Repo artifact account"
+sidebar:
+  nav: setup
+---
+
+{% include toc %}
+
+This shows how to configure a Git Repo artifact account so Spinnaker can use entire repositories as a single artifact.
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
This shows how to configure a Git Repo artifact account so Spinnaker can use an entire repository as a single artifact.
```

Comment (dorbin) [reply setup/artifacts/gitrepo.md]: Unless it's possible to (and plausible that one would want to) have an artifact comprising multiple repos.

File: setup/artifacts/gitrepo.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+```bash
+hal config artifact gitrepo enable
+```
+
+## Configure Auth
+Choose to set up either token, user-password or ssh key auth below.
+
+### Token Auth
+
+Start by generating an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
1. Generate an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.
```

File: setup/artifacts/gitrepo.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Place the token in a file (`$TOKEN_FILE`) readable by Halyard:
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
1. Place the token in a file (`$TOKEN_FILE`) readable by Halyard:
```

File: setup/artifacts/gitrepo.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+echo $TOKEN > $TOKEN_FILE
+```
+
+Add an artifact account:
+
+```bash
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
   ```bash
```

File: setup/artifacts/gitrepo.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
   hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \
```

File: setup/artifacts/gitrepo.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    --token-file $TOKEN_FILE
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
       --token-file $TOKEN_FILE
```

File: setup/artifacts/gitrepo.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+```
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
   ```
```

File: setup/artifacts/gitrepo.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+### Token Auth
+
+Start by generating an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.
+
+Place the token in a file (`$TOKEN_FILE`) readable by Halyard:
+
+```bash
+echo $TOKEN > $TOKEN_FILE
+```
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
   ```
```

File: setup/artifacts/gitrepo.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+Choose to set up either token, user-password or ssh key auth below.
+
+### Token Auth
+
+Start by generating an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.
+
+Place the token in a file (`$TOKEN_FILE`) readable by Halyard:
+
+```bash
+echo $TOKEN > $TOKEN_FILE
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
   echo $TOKEN > $TOKEN_FILE
```

File: setup/artifacts/gitrepo.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+## Configure Auth
+Choose to set up either token, user-password or ssh key auth below.
+
+### Token Auth
+
+Start by generating an access token for your Git provider (eg, [GitHub](https://github.com/settings/tokens){:target="\_blank"} or [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html){:target="\_blank"}). The token requires the __repo__ scope.
+
+Place the token in a file (`$TOKEN_FILE`) readable by Halyard:
+
+```bash
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
   ```bash
```

File: setup/artifacts/gitrepo.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+echo $TOKEN > $TOKEN_FILE
+```
+
+Add an artifact account:
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
1. Add an artifact account:
```

File: setup/artifacts/gitrepo.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \
+    --token-file $TOKEN_FILE
+```
+
+
+### User-Password Auth
+
+The username-password file contents should be in the format:
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md]:
```suggestion
1. Create a username-password file, with contents in the following format:
```

Comment (dorbin) [reply setup/artifacts/gitrepo.md]: (Similarly, number the steps and indent the command so the numbering works.)

File: setup/artifacts/gitrepo.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+hal config artifact gitrepo account add $ARTIFACT_ACCOUNT_NAME \
+    ----username-password-file $PASSWORD_FILE
+```
+
+
+### SSH Key Auth
+
+Add an artifact account:
```
Comment (dorbin) [inline setup/artifacts/gitrepo.md:66]: And no need to number this one, because it's just the one directive.

**Review by dorbin — COMMENTED** [review]
Thanks for this PR, @KathrynLewis ! I made a few suggestions.

File: setup/artifacts/gitrepo.md (hunk 1 above)
Comment (KathrynLewis) [reply setup/artifacts/gitrepo.md]: Nope, you're right, it would be just one 👍

**Review by 317brian — APPROVED** [review]
Thanks for the quick review on this @dorbin

**Pull request merged**